### PR TITLE
Update "PassInfo" for BinTools register support

### PR DESCRIPTION
**Pull request by qubka**
Made for the support of that pull request : https://github.com/alliedmodders/sourcemod/pull/1020

### CODE REVIEW

**Comment by peace-maker** [comment]
This doesn't seem to be the right place for this, since SourceHook is made for hooking virtual functions, which have a fixed calling convention which never passes values in random registers.

**Comment by qubka** [comment]
> This doesn't seem to be the right place for this, since SourceHook is made for hooking virtual functions, which have a fixed calling convention which never passes values in random registers.

I thought to make support fo different callconv, but now i think better to add support for registers like EAX, EDX, ECX, EBX and allow used them for generation of fastcall if user will need, because fastcall can look like different

File: core/sourcehook/sourcehook.h [hunk 1]
```diff
@@ -196,11 +196,34 @@ namespace SourceHook
 			PassFlag_RetMem		= (1<<6),		/**< Object is returned in memory (through hidden first param */
 			PassFlag_RetReg		= (1<<7)		/**< Object is returned in EAX(:EDX) */
 		};
+		
+		enum RegisterType
```
Comment (KyleSanderson) [inline core/sourcehook/sourcehook.h]: This is wrong, this file is dynamically generated. This needs to be in the hxx file.

Comment (qubka) [reply core/sourcehook/sourcehook.h]: Oh thanks for your help, i added changes there and remove other commits

**Comment by dvander** [comment]
I agree with @peace-maker. This doesn't seem necessary unless there's a virtual function in the SDK we're not able to cover.

**Pull request closed without merging**